tests/test.js

```diff
@@ -56,52 +56,37 @@ describe("Embroider build", function () {
 });
 
 describe("When rootURL and publicPath do not match", function () {
-  let indexHtml, manifest, favicons, stylesheets;
+  let indexHtml, stylesheets;
 
   before(async function () {
     indexHtml = await getIndexHtml("rooturl-and-publicpath-differ");
-    favicons = indexHtml.querySelectorAll(
-      'link[rel="icon"], link[rel="apple-touch-icon"]',
-    );
-    manifest = indexHtml.querySelector('link[rel="manifest"]');
     stylesheets = indexHtml.querySelectorAll('link[rel="stylesheet"]');
   });
 
-  it("Favicons have integrity hash", function () {
-    for (const favicon of favicons) {
-      expect(favicon.hasAttribute("integrity")).to.be.true;
-      expect(favicon.getAttribute("integrity")).to.be.a("string");
-    }
-  });
-
-  it("Favicons have crossorigin attribute", function () {
-    for (const favicon of favicons) {
-      expect(favicon.hasAttribute("crossorigin")).to.be.true;
-      expect(favicon.getAttribute("crossorigin")).to.be.a("string");
-      expect(favicon.getAttribute("crossorigin")).to.equal("anonymous");
-    }
-  });
-
-  it("Manifest has integrity hash", function () {
-    expect(manifest.hasAttribute("integrity")).to.be.true;
-    expect(manifest.getAttribute("integrity")).to.be.a("string");
-  });
-
-  it("Manifest has crossorigin attribute", function () {
-    expect(manifest.hasAttribute("crossorigin")).to.be.true;
-    expect(manifest.getAttribute("crossorigin")).to.be.a("string");
-    expect(manifest.getAttribute("crossorigin")).to.equal("anonymous");
-  });
-
   it("Some links start with publicPath, some start with rootURL", function () {
     for (const stylesheet of stylesheets) {
       const href = stylesheet.getAttribute("href");
       expect(href.startsWith("/a-different-public-path/")).to.be.true;
     }
+  });
+});
 
-    for (const favicon of favicons) {
-      const href = favicon.getAttribute("href");
-      expect(href.startsWith("/a-root-url/")).to.be.true;
+describe("Link tags with valid rel attributes are processed", function () {
+  let indexHtml, links;
+  before(async function () {
+    indexHtml = await getIndexHtml("link-rel-without-integrity-hash-support");
+    links = indexHtml.querySelectorAll("link");
+  });
+  it(`Processes link when rel includes 'module', 'modulepreload', or 'stylesheet'`, function () {
+    for (const link of links) {
+      const rel = link.getAttribute("rel");
+      const integrity = link.getAttribute("integrity");
+      if (rel.includes("module") || rel.includes("stylesheet") || rel.includes("modulepreload")) {
```

Comment:
I'm not a big fan of such logic in tests to be honest. It's too easy to have the tests not asserting anymore what is actually wanted. E.g. in this case if there is no `<link>` element anymore at all the tests would still pass. Do you mind refactoring to a more dump but explicit style?

```js
        expect(indexHtml.querySelector("link[rel="stylesheet"]).hasAttribute("integrity")).to.be.true;
expect(indexHtml.querySelector("link[rel="alternate stylesheet"]).hasAttribute("integrity")).to.be.true;
// ...

expect(indexHtml.querySelector("link[rel="icon"]).hasAttribute("integrity")).to.be.false
// ...
```

Additionally it would be great if we could assert for `crossorigin` attribute as well.

Reply:
I'm realizing there's some overlap with the first loop in the test that loops over all of the scenarios. Would you rather have me move these into the `default` test app instead of a new test app, do the assertions as tests in the main loop, and remove the test app I added?

Reply:
That would be great if possible. Less test apps to maintain is always good.